kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/ProcurementCardCreateDocumentServiceImpl.java

```diff
@@ -0,0 +1,448 @@
+
+                transactionLineNumber++;
+            }
+
+            pcardDocument.getFinancialSystemDocumentHeader().setFinancialDocumentTotalAmount(documentTotalAmount);
+            // PCDO Default Description
+            setupDocumentDescription(pcardDocument);
+
+            // Remove duplicate messages from errorText
+            String messages[] = StringUtils.split(errorText, ".");
+            for (int i = 0; i < messages.length; i++) {
+                int countMatches = StringUtils.countMatches(errorText, messages[i]) - 1;
+                errorText = StringUtils.replace(errorText, messages[i] + ".", "", countMatches);
+            }
+            // In case errorText is still too long, truncate it and indicate so.
+            Integer documentExplanationMaxLength = dataDictionaryService.getAttributeMaxLength(DocumentHeader.class.getName(), KFSPropertyConstants.EXPLANATION);
+            if (documentExplanationMaxLength != null && errorText.length() > documentExplanationMaxLength.intValue()) {
+                String truncatedMessage = " ... TRUNCATED.";
+                errorText = errorText.substring(0, documentExplanationMaxLength - truncatedMessage.length()) + truncatedMessage;
+            }
+            pcardDocument.getDocumentHeader().setExplanation(errorText);
+        }
+        catch (WorkflowException e) {
+            LOG.error("Error creating pcdo documents: " + e.getMessage(),e);
+            throw new RuntimeException("Error creating pcdo documents: " + e.getMessage(),e);
+        }
+
+        return pcardDocument;
+    }
+    
+    /**
+     * Creates card holder record and sets that record to the document given.
+     *
+     * @param pcardDocument Procurement card document to place the record in.
+     * @param transaction The transaction to set the card holder record fields from.
+     */
+    //copy-pasted from super class to use correct object types (edu.arizona instead of org.kuali)
+    protected void createCardHolderRecord(ProcurementCardDocument pcardDocument, ProcurementCardTransaction transaction) {
+        ProcurementCardHolder cardHolder = new ProcurementCardHolder();
+
+        cardHolder.setDocumentNumber(pcardDocument.getDocumentNumber());
+        cardHolder.setTransactionCreditCardNumber(transaction.getTransactionCreditCardNumber());
+
+            final ProcurementCardDefault procurementCardDefault = retrieveProcurementCardDefault(transaction.getTransactionCreditCardNumber());
+            if (procurementCardDefault != null) {
+            if (getParameterService().getParameterValueAsBoolean(ProcurementCardCreateDocumentsStep.class, ProcurementCardCreateDocumentsStep.USE_CARD_HOLDER_DEFAULT_PARAMETER_NAME)) {
+                cardHolder.setCardCycleAmountLimit(procurementCardDefault.getCardCycleAmountLimit());
+                cardHolder.setCardCycleVolumeLimit(procurementCardDefault.getCardCycleVolumeLimit());
```

Comment:
Refactoring (or completing the refactor of) the field type should be done on a separate ticket, not here where the primary concern is the routing functionality.